### PR TITLE
Provide a reconcile command to try and bring clusters back to desired…

### PR DESCRIPTION
**Pull request by paulgmiller**
… state

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by paulgmiller** [comment]
@FumingZhang  I didn't use the decorator pattern becauase this reconcile command should never grow or change and when we put it in non - preview we should be able to reuse the exact same function like show. 

This repalces https://github.com/Azure/azure-cli-extensions/pull/4083

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -905,6 +905,16 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
     return aks_update_decorator.update_mc_preview(mc)
 
 
+# Do a get and put to try and bring clusters to a desired state.
+def aks_reconcile(cmd, client, resource_group_name, name, # pylint: disable=unused-argument
+               headers, no_wait):
+    mc = client.get(resource_group_name, name)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:911]:
```suggestion
               aks_custom_headers, no_wait):
    mc = client.get(resource_group_name, name)
    headers = get_aks_custom_headers(aks_custom_headers)
```

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1813,7 +1823,15 @@ def aks_agentpool_get_upgrade_profile(cmd,   # pylint: disable=unused-argument
...
     return client.get_upgrade_profile(resource_group_name, cluster_name, nodepool_name)
-
+    
+def aks_agentpool_reconcile(cmd,     # pylint: disable=unused-argument
+                       client,
+                       resource_group_name,
+                       cluster_name,
+                       nodepool_name, no_wait):
+    ap = client.get(resource_group_name, cluster_name, nodepool_name)
+    return sdk_no_wait(no_wait, client.begin_create_or_update, resource_group_name, cluster_name, nodepool_name, ap)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/custom.py:1833]:
```suggestion
                       nodepool_name, aks_custom_headers, no_wait):
    ap = client.get(resource_group_name, cluster_name, nodepool_name)
    headers = get_aks_custom_headers(aks_custom_headers)
    return sdk_no_wait(no_wait, client.begin_create_or_update, resource_group_name, cluster_name, nodepool_name, ap, headers=headers)
```

**Review by FumingZhang — COMMENTED** [review]
Yeah, I like the idea to use a separate command to do the "empty" PUT request. Left some suggestions to pass the custom header.

**Comment by FumingZhang** [comment]
And from CI [log](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1523181&view=logs&j=ccdac6d3-465a-5ada-8819-d5b4a6b3a631&t=ffdbba7d-e523-5908-c7bf-0703bafb81b0), need to add help message for the new commands.

**Comment by paulgmiller** [comment]
@FumingZhang I pushed some help but style and linter in azdev locally give me alot of failure snot relatedto my change. Is that expected?

**Comment by paulgmiller** [comment]
> Why not `az aks update -n <> -g <>` ? that's what my private alias does

I'm not sure update is better https://github.com/Azure/azure-cli-extensions/pull/4759

**Comment by paulgmiller** [comment]
Moving this to https://github.com/Azure/azure-cli-extensions/pull/4083

**Pull request closed without merging**